### PR TITLE
moved mqtt config into its own node

### PR DESCRIPTION
**Pull request by Dirnei**
# Breaking change!

This change will break the current configuration of the broker node, but this is the first step to enable  the support of other mqtt devices like the [Shelly 2.5](https://shelly.cloud/products/shelly-25-smart-home-automation-relay/).

I also improved the UI of the mqtt configuration to prevent missconfiguration of the protocol.

> The good thing is that it only needs a few clicks to fix!

### CODE REVIEW

**Comment by Dirnei** [comment]
@mdhom is it possible for you to test the phillips hue devices?

**Comment by Dirnei** [comment]
@mdhom please have a look if this is a good solution for this device. I'm not sure if this is intuitive or not

Otherwise it's mostly done. I will begin documentation tomorrow.

**Comment by Dirnei** [comment]
I found a severe bug! I unintentionally removed the ```getDeviceList()``` methode from bridge configuration. No devices are schown in generic-lamp node right now. Old, allready configured lamps are still working but it is not possible to add new ones. Fix is in progress

**Comment by Andreas-Huber** [comment]
Question regarding the breaking changes: Upgrading the config to the new one is probably not easily possible?

**Comment by Dirnei** [comment]
Sadly not, but it takes only a few clicks so its not that bad...

File: docs/nodes/override-status.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Override state
+
+The state can be set to ```ON```, ```OFF``` or ```TOGGLE```. Becarefull with ```TOGGLE``` because when not all Lamps in your flow are not in the same state, the ones that are on turn off and vice versa.
```
Comment (Andreas-Huber) [inline docs/nodes/override-status.md:3]: Does: "not all Lamps in your flow are ~~not~~ in the same state" make more sense?

Comment (Dirnei) [reply docs/nodes/override-status.md]: Totally agree

**Review by Andreas-Huber — APPROVED** [review]
Fixed the things I found myself.

**Pull request merged**